File: shared/dashboard/src/commonMain/kotlin/com/uj/dashboard/Model/DashboardViewData.kt
package com.uj.dashboard.Model

import com.uj.core.viewmodel.BaseViewData

data class DashboardViewData (
    // Header section
    var greetingLabel: String,
    val currentDateLabel: String,
    val daysRemainingInYear: String,
    val daysLeftLabel: String,

    // Self check-in section
    var selfCheckInBtnLabel: String,

    // Toolbox section
    var toolboxHeader:String,
    val toolboxItems: List<ToolboxItem>

) : BaseViewData()

data class ToolboxItem(
    var image: String,
    var name: String
)

File: iosApp/iosApp/Modules/Dashboard/DashboardViewData.swift
//
//  DashboardViewData.swift
//  iosApp
//
//  Created by Ujjwal Chafle on 05/09/2022.
//  Copyright © 2022 orgName. All rights reserved.
//

import Foundation


enum DashboardViewData {
    case empty
    case content(Content)
}


extension DashboardViewData {
    struct Content {
        let greetingLabel: String
        let currentDateLabel: String
        let daysRemainingInYear: String
        let daysLeftLabel: String
        
        // Self check-in section
        let selfCheckInBtnLabel: String
        
        // Toolbox section
        let toolboxHeader:String
        let toolboxItems: [ToolboxItem]
        
        
        struct ToolboxItem {
            let icon: String
            let name: String
        }
    }
    
}
